components/dashboard/src/app/AppRoutes.tsx

```diff
@@ -316,6 +324,7 @@ export const AppRoutes: FunctionComponent<AppRoutesProps> = ({ user, teams }) =>
                     />
                 )}
             </div>
+            {useNewWorkspacesList && <WebsocketClients />}
```

Comment:
I find it a little confusing to have react components like this that return null and do sideffects. Is there any advantage to do itr like this compared to calling a sideffect function in e.g. `useEffect`?

Reply:
Some details in [this comment](https://github.com/gitpod-io/gitpod/pull/15738#discussion_r1072619719) - but it's mainly to encapsulate when this listener gets setup. I wanted it after all of the redirects/alternate rendering flows at the top of this component, so once we render our main router. I could also adjust rendering logic in this component to move the actual ui that we `return` at the end into it's own component, and call the `useListenToWorkspaceWSMessages` hook in there. I went this route as it felt like less of a change, but am fine doing it either way. I also perhaps falsely anticipated that we'd have other similar listeners we'd want to setup for incoming ws messages, and this could be a place to add those as well.